src/c2.c

```diff
@@ -1395,10 +1397,21 @@ static inline void c2_match_once_leaf(session_t *ps, const struct managed_win *w
 			case C2_L_PCLASSG: tgt = w->class_general; break;
 			case C2_L_PCLASSI: tgt = w->class_instance; break;
 			case C2_L_PROLE: tgt = w->role; break;
+			case C2_L_PWMSTATE: {
+				ntargets = 0;
+				for (enum wm_state i = 1; i < NUM_WMSTATES; ++i) {
+					if (win_check_wm_state(w, i)) {
+						targets[ntargets++] = WMSTATES[i];
```

Comment:
This can just be a bitmask, in fact we could just use `w->wm_state`

Reply:
Can you elaborate on what exactly you want to change?

Reply:
Sorry, looks like it's not very doable. 

Basically I was think we could have the `tgt` be the wm_state bitmask, and for matching, we just check if the flag is set. Now I realized this is not how c2 matches things.

If there are more bitmask patterns, maybe we could add a bitmask pattern matching. But if only wm_state would use this, then it's probably not worth it.

Reply:
Yes, we currently only have integer and string comparisons. We could add bitmask checks for exact strings, but I don't think the added complexity is worth it just for `wm_state`. That's why I opted to use the existing string-rules based on the extended array matching.

Reply:
OK, that makes sense.